src/openrct2/peep/GuestPathfinding.cpp

```diff
@@ -2094,11 +2094,24 @@ int32_t guest_path_finding(Guest* peep)
         int32_t select = peep->no_of_rides % numEntranceStations;
         while (select > 0)
         {
-            closestStationNum = bitscanforward(entranceStations);
-            entranceStations &= ~(1 << closestStationNum);
-            select--;
+            for (size_t i = 0; i < entranceStations.size(); i++)
+            {
+                if (entranceStations[i])
+                {
+                    entranceStations[i] = false;
+                    select--;
+                    break;
+                }
+            }
+        }
+        for (size_t i = 0; i < entranceStations.size(); i++)
+        {
+            if (entranceStations[i])
+            {
+                closestStationNum = i;
+                break;
+            }
         }
-        closestStationNum = bitscanforward(entranceStations);
     }
```

Comment:
I think it would be nice if this whole loop and comment above was moved to a new function.